types/index.d.ts

```diff
@@ -263,6 +263,10 @@ declare module "miragejs/-types" {
   /** A marker type for easily constraining type parameters that must be shaped like a Registry */
   export type AnyRegistry = Registry<AnyModels, AnyFactories>;
 
+  type MaybePromise<T> = T | PromiseLike<T>;
+  type ValidResponse = object | number | string | boolean | null
```

Comment:
Instead of `object`, could you please use `Record<PropertyKey, any>`?  `object` in typescript is actually any non-primitive value, which I think is probably not what we want here.

Reply:
I don't think this would be a breaking change, I see it as a bugfix of the types.  Realistically, I wouldn't expect there to be an impact to users, since I'm guessing nobody was relying on an api handler returning a function.  :)